py-rse/errors.Rmd

```diff
@@ -481,17 +555,35 @@ logging.info('This is just for information.')
 WARNING:root:This is a warning.
 ```
 
-We can configure logging to send messages to a file instead of to standard error
+Rewriting the `collate.py` example above using the logging module
+yields code that is less cluttered:
+
+```python
+import logging
+
+
+logging.info('Processing files...')
+for file_name in args.infiles:
+    logging.debug('Reading in {file_name}...')
+    if file_name[-4:] != '.csv':
+        raise OSError(f'{file_name} is not a CSV-file.')
+    with open(file_name, 'r') as reader:
+        logging.debug('Computing word counts...')
+        update_counts(reader, word_counts)
+```
+
```

Comment:
I like this addition of logging statements to `collate.py`.

A few lines later the text talks about defining a single flag `-v` or `--verbose` that changes the logging level from `WARNING` (the default) to `DEBUG` (the noisiest level). To round off this lesson on error handling, I think it would be great to actually add a `-v` option to `collate.py` and run it both in regular and verbose mode to show the different messages, i.e:

```
$ python collate.py ../results/dracula.csv ../results/moby_dick.csv -n 5 
```
then
```
$ python collate.py ../results/dracula.csv ../results/moby_dick.csv -n 5 -v
```

Reply:
I saw [Greg added this as the last exercise](https://github.com/merely-useful/merely-useful.github.io/pull/357/files#diff-f14ced81656f99585a40ffd6bad2abd8R704).